docs/proposals/dns-using-envoy.md

```diff
@@ -0,0 +1,73 @@
+# Use Envoy DNSFilter for DNS resolution
+
+## Context
+
+Currently, Kuma relies on an external DNS to provide VIPs which are then added as services which proxy back to the actual Cluster in Envoy.
+For VIPs to be resolved Kuma CP runs a DNS server which maintains this mapping.
+
+This current setup has some limitations:
+
+- Apps need to set their DNS to point to the CP's DNS (or CNIs need to configure DNS to be correct)
+- The CP needs to run a DNS and handle the load
+- When the CP is down DNS requests will fail. This is usually not the case in a Service Mesh
+- DNS configuration is global to all meshes
+- DNS is a separate component which makes it less customisable compared to things that end up in XDS
+- This DNS is not aware of the origin of requests, therefore isolation of meshes breaks as anyone can resolve any domain
+
+In this proposal we suggest using [Envoy's DNSFilter](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/udp_filters/dns_filter) to replace the CP DNS.
+We add a new `DNS policy` to be able to control the configuration of the DNS.
+
+## Out of scope
+
+* UDP transparent proxying
```

Comment:
Interesting why would one need that? I mean our iptables today do not support any UDP and DNS is working fine (no one interferes with it). Do you expect there might be cases where UDP transparent proxying will be needed?

Reply:
Hmm when I wrote my POC at the beginning I had setup our testing env with UDP iptables to redirect all traffic through the same ports as the TCP transparent proxying. One advantage to this was to be able to pick any address for DNS without needing any setting in my iptables.
Without UDP transparent proxying I add an iptable rule just for the address I've picked.

The TLDR being: It's not needed. I just wanted to at the beginning before changing my mind so I thought I'd put it in the proposal.